modules/social_features/social_group/modules/social_group_request/src/Controller/GroupRequestController.php

```diff
@@ -0,0 +1,184 @@
+    AccountInterface $current_user
+  ) {
+    $this->entityFormBuilder = $entity_form_builder;
+    $this->setMessenger($messenger);
+    $this->cacheTagsInvalidator = $cache_tags_invalidator;
+    $this->setStringTranslation($string_translation);
+    $this->entityTypeManager = $entity_type_manager;
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity.form_builder'),
+      $container->get('messenger'),
+      $container->get('cache_tags.invalidator'),
+      $container->get('string_translation'),
+      $container->get('entity_type.manager'),
+      $container->get('current_user')
+    );
+  }
+
+  /**
+   * Return the title for approve request confirmation page.
+   */
+  public function getTitleApproveRequest(GroupInterface $group, GroupContentInterface $group_content) {
+    return $this->t('Approve membership request for the group @group_title', ['@group_title' => $group->label()]);
+  }
+
+  /**
+   * Return the title for reject request confirmation page.
+   */
+  public function getTitleRejectRequest(GroupInterface $group, GroupContentInterface $group_content) {
+    return $this->t('Reject membership request for the group @group_title', ['@group_title' => $group->label()]);
+  }
+
+  /**
+   * Builds the form to create new membership on membership request approve.
+   */
+  public function approveRequest(GroupInterface $group, GroupContentInterface $group_content) {
+    /** @var \Drupal\group\Plugin\GroupContentEnablerInterface $plugin */
+    $plugin = $group->getGroupType()->getContentPlugin('group_membership');
+
+    // Pre-populate a group membership from Membership request.
+    $group_content = $this->entityTypeManager()->getStorage('group_content')->create([
+      'type' => $plugin->getContentTypeConfigId(),
+      'gid' => $group->id(),
+      'entity_id' => $group_content->getEntity()->id(),
+    ]);
+
+    $this->cacheTagsInvalidator->invalidateTags(['request-membership:' . $group->id()]);
+
+    return $this->entityFormBuilder->getForm($group_content, 'add');
+  }
+
+  /**
+   * Callback to request membership.
+   */
+  public function requestMembership(GroupInterface $group) {
+    $response = new AjaxResponse();
+
+    $contentTypeConfigId = $group
+      ->getGroupType()
+      ->getContentPlugin('group_membership_request')
+      ->getContentTypeConfigId();
+
+    $request = $this->entityTypeManager()->getStorage('group_content')->getQuery()
+      ->condition('type', $contentTypeConfigId)
+      ->condition('gid', $group->id())
+      ->condition('entity_id', $this->currentUser()->id())
+      ->condition('grequest_status', GroupMembershipRequest::REQUEST_PENDING)
+      ->count()
+      ->execute();
+
+    if ($request == 0) {
+      $request_form = $this->entityFormBuilder->getForm(GroupRequestMembershipRequestForm::class, $group);
```

Comment:
1st argument of the **getForm** method should be an entity

Reply:
Fixed